src/applications/personalization/dashboard/components/cerner-widgets.js

```diff
@@ -4,20 +4,31 @@ import { mhvUrl } from 'platform/site-wide/mhv/utilities';
 import { getCernerURL } from 'platform/utilities/cerner';
 
 // Helper component that takes an array of facility names and a separator string and returns some JSX to style the list of facility names.
-const FacilityList = ({ facilities, separator }) => {
-  const newArray = [];
-  // first make an array that alternates between a facility name and the
-  // separator word wrapped in a <span> for styling purposes
-  facilities.forEach(el => {
-    newArray.push(el);
-    newArray.push(
+const FacilityList = ({ facilities, separator, finalSeparator }) => {
+  let newArray = [];
+  if (facilities.length > 1) {
+    const lastElement = facilities.pop();
+    const secondToLastElement = facilities.pop();
+    const finalSeparatorSpan = (
       <span className="vads-u-font-weight--normal vads-u-font-size--base">
-        {separator}
-      </span>,
+        {finalSeparator || separator}
+      </span>
     );
-  });
-  // we don't need the last separator in the array
-  newArray.pop();
+    // first make an array that alternates between a facility name and the
+    // separator word wrapped in a <span> for styling purposes
+    facilities.forEach(el => {
+      newArray.push(el);
+      newArray.push(
+        <span className="vads-u-font-weight--normal vads-u-font-size--base">
+          {separator}
+        </span>,
+      );
+    });
+    newArray.push(secondToLastElement, finalSeparatorSpan, lastElement);
+  } else {
+    newArray = newArray.concat(facilities);
+  }
+
```

Comment:
I think this works as it is, but thought maybe in the future we could look at refactoring a bit.

My idea would be to just use the main map and avoid any array manipulation with pop and push. Something like this I think would work equally well:

```
const SeperatorSpan = ({ children }) => (
  <span className="vads-u-font-weight--normal vads-u-font-size--base">
    {children}
  </span>
);

SeperatorSpan.propTypes = {
  children: PropTypes.node,
};

// Helper component that takes an array of facility names and a separator string and returns some JSX to style the list of facility names.
const FacilityList = ({ facilities, separator, finalSeparator }) => {
  return (
    <>
      {facilities.map((el, i, orig) => {
        return (
          <strong key={i}>
            {el}
            {el !== orig.at(-2) &&
              el !== orig.at(-1) && <SeperatorSpan>{separator}</SeperatorSpan>}
            {el === orig.at(-2) && (
              <SeperatorSpan>{finalSeparator}</SeperatorSpan>
            )}
          </strong>
        );
      })}
    </>
  );
};
```

Just a thought for future refactors. What do you think?

Reply:
Yeah I think I agree. In order to reduce technical debt I will update it now. I was thinking of doing something similar to this, but got carried away on my other route. Thanks!